python/hsfs/storage_connector.py

```diff
@@ -652,6 +661,8 @@ def spark_options(self):
             props["sfToken"] = self._token
         if self._warehouse is not None:
             props["sfWarehouse"] = self._warehouse
+        if self._application is not None:
```

Comment:
keep just `if self._application:` - in python empty string evaluates as false, and we don't want to set the field if the string is None or empty